lxd/profiles.go

```diff
@@ -245,24 +272,39 @@ func profilesPost(d *Daemon, r *http.Request) response.Response {
 
 	// Update DB entry.
 	err = d.db.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
-		devices, err := db.APIToDevices(req.Devices)
+		devices, err := dbCluster.APIToDevices(req.Devices)
 		if err != nil {
 			return err
 		}
 
-		current, _ := tx.GetProfile(projectName, req.Name)
+		current, _ := dbCluster.GetProfile(ctx, tx.Tx(), projectName, req.Name)
 		if current != nil {
 			return fmt.Errorf("The profile already exists")
 		}
 
-		profile := db.Profile{
+		profile := dbCluster.Profile{
 			Project:     projectName,
 			Name:        req.Name,
 			Description: req.Description,
-			Config:      req.Config,
-			Devices:     devices,
 		}
-		_, err = tx.CreateProfile(profile)
+		id, err := dbCluster.CreateProfile(ctx, tx.Tx(), profile)
+		if err != nil {
+			return err
+		}
+
+		err = dbCluster.CreateProfileConfig(ctx, tx.Tx(), id, req.Config)
+		if err != nil {
+			return err
+		}
+
+		for _, device := range devices {
+			err = dbCluster.CreateProfileDevice(ctx, tx.Tx(), id, device)
```

Comment:
@masnax it strikes me as odd/inconsistent that we have only a single `CreateProfileDevice` function, but have support for updating multiple profile devices via `UpdateProfileDevices`. I would have expected there to be a `CreateProfileDevices` instead of `CreateProfileDevice` (even if it does the same thing as it does now internally).

Is there a reason for this, and has this asymmetric pattern been repeated elsewhere?

Reply:
Yep, this definitely should be more consistent. I don't recall exactly why I did it like this, but grepping through the codebase I don't see any reason not to make it more consistent.

Reply:
I'll make this an additional PR since it's a bit out of the scope of this PR.